test/cpp/end2end/ssl_credentials_test.cc

```diff
@@ -77,96 +129,297 @@ class SslCredentialsTest : public ::testing::Test {
     }
   }
 
+  absl::StatusOr<std::shared_ptr<const AuthContext>> DoRpc(
+      const SslCredentialsOptions& options, grpc_ssl_session_cache* cache,
+      bool override_ssl_target_name = true) {
+    ChannelArguments channel_args;
+    if (GetParam().use_session_cache) {
+      channel_args.SetPointer(std::string(GRPC_SSL_SESSION_CACHE_ARG), cache);
+    }
+    if (override_ssl_target_name) {
+      channel_args.SetSslTargetNameOverride("foo.test.google.fr");
+    }
+
+    grpc_ssl_pem_key_cert_pair pem_key_cert_pair = {
+        options.pem_private_key.c_str(), options.pem_cert_chain.c_str()};
+    grpc_channel_credentials* c_creds = grpc_ssl_credentials_create(
+        options.pem_root_certs.empty() ? nullptr
+                                       : options.pem_root_certs.c_str(),
+        options.pem_private_key.empty() ? nullptr : &pem_key_cert_pair, nullptr,
+        nullptr);
+    grpc_ssl_credentials* ssl_creds =
+        reinterpret_cast<grpc_ssl_credentials*>(c_creds);
+    ssl_creds->set_min_tls_version(GetParam().tls_version);
+    ssl_creds->set_max_tls_version(GetParam().tls_version);
+    auto creds = std::make_shared<WrappedChannelCredentials>(ssl_creds);
```

Comment:
Why is this now using the C-core API instead of the public C++ API?

Reply:
The test needs to be parametrized on the TLS version to get better coverage, but the decision was made a while ago to not expose the min/max TLS version setters for the C++-layer SSL credentials (only for TLS credentials).

If you'd prefer, I'm happy to use the public C++ API in the TLS 1.3 case, and only revert to the C-core API in the TLS 1.2 case.

Reply:
This is an e2e test for the C++ SSL creds API.  If a piece of functionality isn't exposed in that API, why are we testing it here?  It seems like if you want to test the underlying C-core API, that should be done in a C-core test instead.

Reply:
Totally fair. Switched.